Dockerfile

```diff
@@ -0,0 +1,18 @@
+# Reference: https://typeofnan.dev/how-to-serve-a-react-app-with-nginx-in-docker/
+
+FROM nginx:alpine
+
+# Copy nginx config file
+COPY nginx/nginx.conf /etc/nginx/conf.d/default.conf
+
+# Set working directory to nginx asset directory
+WORKDIR /usr/share/nginx/html
+
+# Remove default nginx static assets
+RUN rm -rf ./*
+
+# Copy already built static assets
+COPY build .
```

Comment:
Does this command mean we need to first manually build the front end project and then copy the static files built to the container? If it is the case then the host server still needs to install `Node.js` environment which I think is not necessary.
IMO we can use `multi-stage build` feature to build the front end project and inside the container and then copy the build result to the `nginx` container and then run the app there.

Reply:
Yes, you are right that this relies on the `build` directory being ready. I was thinking that CircleCI would build the app and push it to Github, but your solution is even better. I'll do that.